### DISCUSSION TITLE
help to solve this roslyn parser issue.

### DISCUSSION THREAD

**Discussion by ketanmehra30** [dotnet/roslyn#66185] [Q&A]
I am new to roslyn and building a C# parser using it and I want to get the all the field definitions and field references in a code file, I am able to get field definition by traversing through the tree and finding type FieldDeclarationSyntax but i am not able to find corresponding field reference. somone suggested to get the IdentifierNameSyntax node but that returned multiple unnecessary nodes. Any help will be much appreciated.

![ikks64fbal9a1](https://user-images.githubusercontent.com/110596316/210210590-8ab12816-d2e5-4a6b-b598-12237313bbfa.png)

**Comment by CyrusNajmabadi** [comment]
You can use SymbolFinder.FindReferencesAsync for this. 

If you want to do it manually yourself, you need to walk the identifiers and see if they are a reference to your fields.  You can use semanticModel.GetSymbolInfo for this. 

I recommend reading these docs: https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/

Read the ones on syntax and semantics. 

Finally, I'd recommend the Roslyn channel in discord.gg/csharp for general questions.
